pkg/helm/actions/upgrade_release_test.go

```diff
@@ -0,0 +1,183 @@
+				Log:          func(format string, v ...interface{}) {},
+			}
+
+			r := release.Release{
+				Name:      "test",
+				Namespace: "test-namespace",
+				Info: &release.Info{
+					FirstDeployed: time.Time{},
+					Status:        "deployed",
+				},
+				Version: 1,
+				Chart: &chart.Chart{
+					Metadata: &chart.Metadata{
+						Name:    "influxdb",
+						Version: "3.0.2",
+					},
+				},
+			}
+
+			store.Create(&r)
+
+			rel, err := UpgradeRelease("test-namespace", "test", tt.chartPath, nil, actionConfig)
+			if err == nil && tt.err != nil {
+				t.Error(err)
+			}
+			if err != nil && err.Error() != tt.err.Error() {
+				t.Error("Error occurred while installing chartPath")
+			}
+			if rel != nil {
+				if rel.Name != "test" {
+					t.Error("Release releaseName isn't matching")
+				}
+				if rel.Namespace != "test-namespace" {
+					t.Error("Namespace releaseName isn't matching")
+				}
+				if rel.Info.Status != release.StatusDeployed {
+					t.Error("Chart status should be deployed")
+				}
+				if rel.Chart.Metadata.Name != tt.chartName {
+					t.Error("Chart name mismatch")
+				}
+				if rel.Chart.Metadata.Version != tt.chartVersion {
+					t.Error("Chart version mismatch")
+				}
+				if rel.Version != 2 {
+					t.Error("Upgrade should increase the version count")
+				}
+			}
+		})
+	}
+}
+
+func TestUpgradeReleaseWithCustomValues(t *testing.T) {
+	tests := []struct {
+		releaseName  string
+		chartPath    string
+		chartName    string
+		chartVersion string
+		values       map[string]interface{}
+		err          error
+	}{
+		{
+			releaseName:  "upgrade valid release with custom values should return successful response",
+			chartPath:    "../testdata/influxdb-3.0.2.tgz",
+			chartName:    "influxdb",
+			chartVersion: "3.0.2",
+			values: map[string]interface{}{
+				"service": map[string]interface{}{"type": "NodePort"},
+			},
+			err: nil,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.releaseName, func(t *testing.T) {
+			store := storage.Init(driver.NewMemory())
+			actionConfig := &action.Configuration{
+				Releases:     store,
+				KubeClient:   &kubefake.PrintingKubeClient{Out: ioutil.Discard},
+				Capabilities: chartutil.DefaultCapabilities,
+				Log:          func(format string, v ...interface{}) {},
+			}
+
+			r := release.Release{
+				Name:      "test",
+				Namespace: "test-namespace",
+				Info: &release.Info{
+					FirstDeployed: time.Time{},
+					Status:        "deployed",
+				},
+				Version: 1,
+				Chart: &chart.Chart{
+					Metadata: &chart.Metadata{
+						Name:    "influxdb",
+						Version: "3.0.2",
+					},
+				},
+			}
+
+			store.Create(&r)
+
+			rel, err := UpgradeRelease("test-namespace", "test", tt.chartPath, tt.values, actionConfig)
+			if err == nil && tt.err != nil {
+				t.Error(err)
+			}
+			if err != nil && err.Error() != tt.err.Error() {
+				t.Error("Error occurred while installing chartPath")
+			}
+			if rel != nil {
+				if rel.Name != "test" {
```

Comment:
same as above.

Reply:
@pedjak do you renameing `rel` to `releaseName` ?

Reply:
Done